.travis.yml

```diff
@@ -8,16 +8,29 @@ services:
 
 if: type != pull_request  # do not build anything for PRs
 
-stages:
+
+stages: 
+- name: build_image
 - name: test
+- name: push_image
+  if: branch = master
 
 jobs:
   include:
+  - stage: build_image
```

Comment:
To keep the CI config simple, I think for all intents and purposes, building and running the Docker image is a test, too, so I would add it to the test stage, at least for that.